### ISSUE TITLE
Update the webapp manifest with a plugin

### ISSUE THREAD

**Issue by kimsible** [Chocobozzz/PeerTube#2689]
I realized that the only way to update the webapp manifest here `client/dist/manifest.webmanifest` was to edit it manually on the server.

So I made this plugin which can edit all the manifest fields including icons : https://github.com/kimsible/peertube-plugin-webapp-manifest

The client part generates the input files and catch the others values on the fly and post them to the routes :
```
POST /plugins/peertube-plugin-webapp-manifest/router/icons
```
=> will place icons to `/storage/plugins/assets/:PLUGIN_NPM_NAME/icons/:ICON_NAME.png` and served them via the router plugin on `/plugins/peertube-plugin-webapp-manifest/router`

```
POST /plugins/peertube-plugin-webapp-manifest/router/manifest
```
=> will update manifest with all new values including icons

The icons are persistent, stored in a directory where they will not be destroyed after uninstalling the plugin.

Every time the app restart (or after upgrading PeerTube), the plugins are registered again, so the manifest is also updated at this time with existing values in the plugin settings.

One thing I don't understand is the "unregistered" function server-side, I try to use it to restore the initial values of the manifest from `client/src/manifest.webmanifest` when uninstalling the plugin but it does not seem to work, any idea @Chocobozzz  ?

<div>
<img width=49% alt=webapp-manifest-1 src=https://user-images.githubusercontent.com/1877318/80380127-6a21d080-889f-11ea-8a3d-9b43d6d8ac21.png />
<img width=49% alt=webapp-manifest-2 src=https://user-images.githubusercontent.com/1877318/80380125-69893a00-889f-11ea-85d5-559793e4531e.png />
</div>

![webapp-manifest-3](https://user-images.githubusercontent.com/1877318/80380721-3d21ed80-88a0-11ea-9fc5-789932634ddf.png)

**Comment by Chocobozzz** [comment]
The `unregister` function should be called by peertube here: https://github.com/Chocobozzz/PeerTube/blob/develop/server/lib/plugins/plugin-manager.ts#L258

Try to put a log in your plugin to see if the function is called

**Comment by kimsible** [comment]
It now seems to work, I updated the plugin. I tryied to use `getRouter` in `unregister` parameters but this param is unavailable for this function, it was to check if PeerTube was at least 2.2 during its development. So `unregister` was not run.

I now use the engine limitation version in `package.json` : https://github.com/kimsible/peertube-plugin-webapp-manifest/blob/master/package.json#L27

I think this issue can be closed =).
